From 2658e2f12b660e81f8baccd98bcf41a693145ed6 Mon Sep 17 00:00:00 2001
From: Rubin Xu <rubinxu@google.com>
Date: Tue, 3 Jun 2025 15:14:01 +0100
Subject: [PATCH] DevicePolicyManager: ignore invalid proxy settings

Ignore exceptions thrown when invalid proxy settings are being
applied, in order to avoid breaking callers performing potentially
security-sensitive operations (like removing an ActiveAdmin)

Bug: 365975561
Test: manual
Flag: EXEMPT bugfix
Merged-In: I496b9ce7a0bbb1e6c245682be57b7357bb432d42
Change-Id: I496b9ce7a0bbb1e6c245682be57b7357bb432d42
(cherry picked from commit 2ea32931ef9a06a762b8cf21584f156140672472)
---
 .../devicepolicy/DevicePolicyManagerService.java    | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
index 964826d1ee73..abd959e102f4 100644
--- a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
+++ b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
@@ -8708,10 +8708,15 @@ public class DevicePolicyManagerService extends IDevicePolicyManager.Stub {
             Slogf.e(LOG_TAG, "Invalid proxy properties, ignoring: " + proxyProperties.toString());
             return;
         }
-        mInjector.settingsGlobalPutString(Global.GLOBAL_HTTP_PROXY_HOST, data[0]);
-        mInjector.settingsGlobalPutInt(Global.GLOBAL_HTTP_PROXY_PORT, proxyPort);
-        mInjector.settingsGlobalPutString(Global.GLOBAL_HTTP_PROXY_EXCLUSION_LIST,
-                exclusionList);
+        try {
+            mInjector.settingsGlobalPutString(Global.GLOBAL_HTTP_PROXY_HOST, data[0]);
+            mInjector.settingsGlobalPutInt(Global.GLOBAL_HTTP_PROXY_PORT, proxyPort);
+            mInjector.settingsGlobalPutString(Global.GLOBAL_HTTP_PROXY_EXCLUSION_LIST,
+                    exclusionList);
+        } catch (Exception e) {
+            //Ignore any potential errors from SettingsProvider, b/365975561
+            Slogf.w(LOG_TAG, "Fail to save proxy", e);
+        }
     }
 
     /**
-- 
2.50.1.351.gcf940e82a1-goog

